swarm_dial.go

```diff
@@ -248,7 +248,7 @@ func (s *Swarm) doDial(ctx context.Context, p peer.ID) (*Conn, error) {
 			log.Debugf("ignoring dial error because we have a connection: %s", err)
 			return conn, nil
 		}
-		if err != context.Canceled {
+		if ctx.Err() == nil {
```

Comment:
Previous condition was correct; we don't want to add a backoff if the user cancelled the context.

Reply:
That's precisely what the new code does. The old code checks for only one of the two (currently) possible context completion errors, that may not have been triggered by the user's context.

Reply:
That’s because we *want* to add the backoff if the deadline is exceeded (ie. timeout), but not if the user manually cancels the context.

Reply:
I've removed this change.